### DISCUSSION TITLE
Simplification for Node TW server when it has host aliases?

### DISCUSSION THREAD

**Discussion by Ca1vino** [TiddlyWiki/TiddlyWiki5#5892] [Q&A]
You more experienced Node TW server users might have an idea or suggestion about this.

I've got a setup with multiple TW servers behind a reverse proxy. The wikis are distinguished by adding a path, like "mywiki.com/wikiname".

For this to work I need a tiddler $:/config/tiddlyweb/host AND to add a server parameter "path-prefix=/wikiname".

Now, if $:/config/tiddlyweb/host tiddler is NOT present, the host part of the URL works fine, the main page displays OK, but the XHR back from the browser to the server is not correct.  It is missing the path (/wikiname in this example).

QUESTION--  Can I modify the code somewhere, so that the XHR path from client back to server is taken from the server's path-prefix parameter when $:/config/tiddlyweb/host is not present?  

The advantage, why I want to do this, is that it makes it possible for a couple of DIFFERENT host aliases to be handled by the SAME tiddlywiki server and the SAME Apache reverse proxy configuration.  This means one less TW5 run-time server and simpler configuration in the reverse proxy.

As it stands now, I need TWO TW servers running depending on the host alias. Otherwise "test.mywiki.com/wikiname" and "mywiki.com/wikiname" could go to the SAME TW server, I'd only need one.

**Comment by pmario** [comment]
The default TW servers CAN NOT handle more than 1 wiki at a time.

**Comment by Ca1vino** [comment]
Do you mean that more than one TW server cannot at the same time use the same file store and expect good results? I have not noticed any problems yet but understand there could be race conditions.
-------- Original message --------From: Mario Pietsch ***@***.***> Date: 18/7/21  2:42 am  (GMT-08:00) To: Jermolene/TiddlyWiki5 ***@***.***> Cc: Ca1vino ***@***.***>, Author ***@***.***> Subject: Re: [Jermolene/TiddlyWiki5] Simplification for Node TW server when it has host aliases? (#5892) 
The default TW servers CAN NOT handle more than 1 wiki at a time.

**Comment by Jermolene** [comment]
Hi @Ca1vino 

> QUESTION-- Can I modify the code somewhere, so that the XHR path from client back to server is taken from the server's path-prefix parameter when $:/config/tiddlyweb/host is not present?

As you've discovered, at the moment the `path-prefix` server variable and `$:/config/tiddlyweb/host` are entirely separate settings. That's simply because those features are implemented via two distinct mechanisms.

> As it stands now, I need TWO TW servers running depending on the host alias. Otherwise "test.mywiki.com/wikiname" and "mywiki.com/wikiname" could go to the SAME TW server, I'd only need one.

Can you configure test.mywiki.com/wikiname to redirect to mywiki.com/wikiname or vice versa?
